### PR TITLE
Partial revert Sharedsnapshot refactor PR

### PR DESCRIPTION
**Pull request by weinan003**
In patch, cfc911d846fa5bf6bfa0b70eee406ae3c98460ca, we create a DSM for sharedsnapshot synchronization between writer and reader. the DSM lifecycle is the same as a session, which we use `dsm_ping_maping` to mark `dsm_segment`. 

However, in centos 6, the DSM will randomly be free after some test cases even the session does not exist. It seems to related to dsm itself. 

Since one pipeline job(icw_gporca_ictcp_centos6) is failed every time, revert this part of code. We still keep dump sharedsnapshot to dsm for cursor use.

## Here are some reminders before you submit the pull request
- [ ] Add tests for the change
- [ ] Document changes
- [ ] Communicate in the mailing list if needed
- [x] Pass `make installcheck`
- [ ] Review a PR in return to support the community

### CODE REVIEW

File: src/backend/utils/time/sharedsnapshot.c [hunk 1]
```diff
@@ -850,17 +786,23 @@ readSharedLocalSnapshot_forCursor(Snapshot snapshot, DtxContext distributedTrans
 void
 AtEOXact_SharedSnapshot(void)
 {
-	ListCell *lc;
-	List *oldlist;
-
-	oldlist = shared_snapshot_files;
-	shared_snapshot_files = NIL;
+	dumpHtab = NULL;
```
Comment (zhangh43) [inline src/backend/utils/time/sharedsnapshot.c:788]: when do you hash_destroy the dumpHtab?

File: src/backend/utils/time/sharedsnapshot.c [hunk 2]
```diff
@@ -850,17 +786,23 @@ readSharedLocalSnapshot_forCursor(Snapshot snapshot, DtxContext distributedTrans
...
 
-	foreach(lc, oldlist)
+	if (created_dump)
 	{
-		BufFileClose((BufFile * ) lfirst(lc));
+		LWLockAcquire(SharedLocalSnapshotSlot->slotLock, LW_EXCLUSIVE);
+
+		ResourceOwnerRelease(TopTransactionResourceOwner,
```
Comment (zhangh43) [inline src/backend/utils/time/sharedsnapshot.c:794]: why call `ResourceOwnerRelease ` at here?

File: src/backend/utils/time/sharedsnapshot.c (hunk 1 above)
Comment (weinan003) [reply src/backend/utils/time/sharedsnapshot.c:788]: the dumpHtab is bind to TopTransactionContext. the context will help us free it.

File: src/backend/utils/time/sharedsnapshot.c (hunk 2 above)
Comment (weinan003) [reply src/backend/utils/time/sharedsnapshot.c:794]: Oh yes, it supposes to be DumpResOwner. I want to free every dump dsm after a transaction finish. I will fix this and add some comments

**Pull request merged**